packages/plugin-stage/sources/drivers/GitDriver.ts

```diff
@@ -7,13 +7,107 @@ import * as stageUtils from '../stageUtils';
 const MESSAGE_MARKER = `Commit generated via \`yarn stage\``;
 const COMMIT_DEPTH = 11;
 
-async function genCommitMessage(cwd: string) {
+
+async function genCommitMessage(cwd: string, changes: Array<stageUtils.FileAction>) {
+  const updates = new Map();
+  const removes = new Map();
+  const adds = new Map();
+  const createdPackages = [];
+  const removedPackages = [];
+
+  for(const change of changes) {
+    const { action, path } = change;
+    const localPath = NodeFS.fromPortablePath(path);
+    const relativePath = posix.relative(cwd, localPath);
+    if (action === stageUtils.ActionType.MODIFY) {
+      const prevManifest = await Manifest.fromText(`git`, [`show`, `HEAD~1:${relativePath}`], {cwd, strict: true});
```

Comment:
Sorry, I wasn't clear - I meant this (pseudocode):

```ts
const {stdout: prevSource} = await execUtils.execvp(`git`, [...], {cwd, strict: true});
const prevManifest = await Manifest.fromText(prevSource, {cwd, strict: true});
```

The `fromText` should take the text buffer in parameter, not a process to spawn.